modules/json/src/lib/parse-json-in-batches.js

```diff
@@ -20,9 +21,21 @@ export default async function* parseJSONInBatches(asyncIterator, options) {
 
   for await (const chunk of asyncIterator) {
     const rows = parser.write(chunk);
-    const jsonPath = parser.getJsonPath().toString();
 
-    if (isFirstChunk) {
+    const jsonpath = rows.length > 0 && parser.getStreamingJsonPath().toString();
+
+    if (rows.length > 0 && isFirstChunk) {
+      if (metadata) {
```

Comment:
`partial-result` | `full-result` were sent before even if the metadata flag wasn't set. How would one consume the batches if those aren't sent (and metadata isn't set to `true`)?

Reply:
Previously you needed to set the `json._rootObjectBatches` option (still supported but no longer documented).

Now setting the top-level `metadata` option (which already triggers an extra initial `metadata` batch) now also triggers these batches. Basically since the app is indicating that it is checking the `batchType` field, it is OK to send additional batches no need for an underscore option.

Reply:
Also note the new batches have cleaner type names.